### PR TITLE
[ENG-11467][create-eas-build-function] improve readme in custom build function templates

### PR DESCRIPTION
**Pull request by szdziedzic**
# Why

Our current readme is outdated and doesn't reference https://docs.expo.dev/custom-builds/functions/ tutorial (which is the best way to start IMO)

# How

Update readme. Make it more "cheerful" and reference our tutorial.

# Test Plan

@sjchmiela code review and him verifying that this is actually user friendly 😅

### CODE REVIEW

File: packages/create-eas-build-function/templates/typescript/package.json [hunk 1]
```diff
@@ -12,6 +12,6 @@
   "devDependencies": {
     "@types/node": "20.11.0",
     "typescript": "^5.3.3",
-    "@expo/steps": "^1.0.55"
```
Comment (sjchmiela) [inline packages/create-eas-build-function/templates/typescript/package.json:15]: Oh gosh, here too. Should we add a script to release-it to update this automatically on a new release?

File: packages/create-eas-build-function/templates/javascript/README.md [hunk 2]
```diff
@@ -1,34 +1,13 @@
...
 
-## How to use it
+This is a custom build function that can be used as a step in the EAS Build custom build process. If you don't know what custom builds are or how to use them, refer to the [custom builds documentation](https://docs.expo.dev/custom-builds/get-started/).
 
-1. Install dependencies: `npm install`.
-2. Implement your function in `src/index.js`.
-3. Install [`ncc`](https://github.com/vercel/ncc) if not yet installed: `npm install -g @vercel/ncc`.
-4. Compile the function with `ncc` by running `npm run build`. Ensure that the `build` directory is not ignored by `.gitignore` / `.easignore`, it must be included in the [archive uploaded when running `eas build`](https://expo.fyi/eas-build-archive).
-5. Use the function in a custom build YAML config. For example:
+This is a way of executing your custom and very own JavaScript code as a part of your build process running on EAS servers.
```
Comment (sjchmiela) [inline packages/create-eas-build-function/templates/javascript/README.md:7]:
```suggestion
This is a way of executing your custom JavaScript code as a part of your build process running on EAS servers.
```

File: packages/create-eas-build-function/templates/javascript/README.md [hunk 3]
```diff
@@ -1,34 +1,13 @@
-# My EAS Build function
+# This is your JavaScript function for EAS Build custom builds 🎉
 
-This is an EAS Build function written in JavaScript. It can be used as a step in a [custom build](https://docs.expo.dev/preview/custom-build-config/).
+## What is this? 👀
 
-## How to use it
+This is a custom build function that can be used as a step in the EAS Build custom build process. If you don't know what custom builds are or how to use them, refer to the [custom builds documentation](https://docs.expo.dev/custom-builds/get-started/).
```
Comment (sjchmiela) [inline packages/create-eas-build-function/templates/javascript/README.md:5]:
```suggestion
This is a custom build function that can be used as a step in the EAS Build build process. If you don't know what custom builds are or how to use them, refer to the [custom builds documentation](https://docs.expo.dev/custom-builds/get-started/).
```

File: packages/create-eas-build-function/templates/javascript/README.md [hunk 4]
```diff
@@ -1,34 +1,13 @@
...
-                name: Bye!
-                command: echo "Bye! The custom function has finished its job."
+## How can I use it? 🚀
 
-    functions:
-        my_function:
-            name: my-function
-            path: ./my-function # The path is resolved relative to this config file.
-    ```
+Please refer to [this](https://docs.expo.dev/custom-builds/functions/) tutorial to learn how to use this function in your EAS Build custom build process.
```
Comment (sjchmiela) [inline packages/create-eas-build-function/templates/javascript/README.md:11]: Should we make whole "this tutorial" a link?

File: packages/create-eas-build-function/templates/javascript/README.md [hunk 5]
```diff
@@ -1,34 +1,13 @@
...
 
-## Learn more
-
-Refer to the [custom builds documentation](https://docs.expo.dev/preview/custom-build-config/).
\ No newline at end of file
+You can also check the [custom build's **config.yml** schema reference](https://docs.expo.dev/custom-builds/schema/) to learn more!
```
Comment (sjchmiela) [inline packages/create-eas-build-function/templates/javascript/README.md:13]:
```suggestion
Check out the [custom build's **config.yml** schema reference](https://docs.expo.dev/custom-builds/schema/) to learn more!
```

**Review by sjchmiela — APPROVED** [review]
Minor comments, applicable to the TypeScript readme too

File: packages/create-eas-build-function/templates/typescript/package.json (hunk 1 above)
Comment (szdziedzic) [reply packages/create-eas-build-function/templates/typescript/package.json:15]: yeah I think so, but I believe we should do it in separate PR

**Pull request merged**